Orchestrator.py

```diff
@@ -42,14 +42,19 @@ def __init__(self, numRobots, initX, initY):
             (
                 i,
                 {
-                   'x':       initX,
-                   'y':       initY,
-                   'heading': 0,
-                   'speed':   0,
+                   'x':                initX,
+                   'y':                initY,
+                   'heading':          0,
+                   'speed':            0,
+                   'movementDuration': 0
                 }
-            ) for i in range(1,self.numRobots+1)
+            ) for i in range(1, self.numRobots+1)
         ])
 
+        # initial movements
+        for dotBotId in range(1,self.numRobots+1):
+            self._updateMovement(source=dotBotId)
```

Comment:
source isn't the right param name! Your function is called `_updateMovement`, its param should be called `dotBotId`. "source" makes zero sense.
Also, you don't have to specify the name of teh param for each function call. So actually remove the `dotBotId=`